src/components/cc-popover/cc-popover.js

```diff
@@ -0,0 +1,263 @@
+        this.close();
+      }
+    });
+
+    this._onEscapeKeyHandler = new EventHandler(this, 'keydown', (event) => {
+      if (event.key === 'Escape') {
+        this.close();
+      }
+    });
+
+    let lastOpenedPopover = null;
+    this._onCcPopoverOpen = new EventHandler(window, 'cc-popover:open', (event) => {
+      // we cannot use event.target because events that happen in shadow DOM have the host element as the target,
+      // when caught outside the component.
+      const popover = event.composedPath()[0];
+
+      if (popover !== this) {
+        lastOpenedPopover = popover;
+      }
+      else {
+        lastOpenedPopover?.close(false);
+        lastOpenedPopover = null;
+      }
+    });
+  }
+
+  // region Public methods
+
+  /**
+   * Opens the popover.
+   */
+  open () {
+    if (!this.isOpen) {
+      this.isOpen = true;
+      dispatchCustomEvent(this, 'open');
+    }
+  }
+
+  /**
+   * Closes the popover.
+   * @param {boolean} [shouldFocus = true] Whereas the button should be focused. This applies only if the popover was opened.
+   */
+  close (shouldFocus = true) {
+    if (this.isOpen) {
+      this.isOpen = false;
+      if (shouldFocus) {
+        this.focus();
+      }
+      dispatchCustomEvent(this, 'close');
+    }
+  }
+
+  /**
+   * Toggle the popover display.
+   */
+  toggle () {
+    if (this.isOpen) {
+      this.close();
+    }
+    else {
+      this.open();
+    }
+  }
+
+  /**
+   * Moves the focus on the button.
+   */
+  focus () {
+    this._buttonRef.value?.focus();
+  }
+
+  // endregion
+
+  // region Lit lifecycle
+
+  updated (changedProperties) {
+    if (changedProperties.has('isOpen')) {
+      if (this.isOpen) {
+        this._onOutsideClickHandler.connect();
+        this._onEscapeKeyHandler.connect();
+      }
+      else {
+        this._onOutsideClickHandler.disconnect();
+        this._onEscapeKeyHandler.disconnect();
+      }
+    }
+  }
+
+  connectedCallback () {
+    super.connectedCallback();
+    this._onCcPopoverOpen.connect();
+  }
+
+  disconnectedCallback () {
+    super.disconnectedCallback();
+    this._onOutsideClickHandler.disconnect();
+    this._onEscapeKeyHandler.disconnect();
+    this._onCcPopoverOpen.disconnect();
+  }
+
+  // endregion
+
+  render () {
+    return html`
+      <div class="wrapper">
+        <cc-button
+          ${ref(this._buttonRef)}
+          .a11yExpanded=${this.isOpen}
+          .accessibleName=${this.accessibleName}
+          ?hide-text=${this.hideText}
+          .icon=${this.icon}
+          @cc-button:click=${this.toggle}
+        >
+          <slot name="button"></slot>
+        </cc-button>
+  
+        ${this.isOpen ? html`
+          <div class="content ${this.position.replace('-', ' ')}" ${ref(this._contentRef)}>
+            <slot></slot>
+          </div>
+        ` : null}
+      </div>
+    `;
+  }
+
+  static get styles () {
+    return [
+      // language=CSS
+      css`
+        :host {
+          display: block;
+
```

Comment:
nitpick: should we remove this empty line or is it the formatter?

Reply:
No, we can't. Stylelint you know...